warehouse/query-core/src/main/java/datawave/query/index/lookup/RangeStream.java

```diff
@@ -250,17 +250,22 @@ public Iterator<QueryPlan> iterator() {
                         queryStream = streams.iterator().next();
                     }
                 }
-                if (queryStream.context() == StreamContext.VARIABLE) {
-                    if (queryStream instanceof Union) {
-                        context = StreamContext.ABSENT;
-                        queryStream = null;
-                        itr = Collections.emptyIterator();
-                        return itr;
-                    } else {
-                        context = StreamContext.PRESENT;
-                    }
-                } else {
-                    context = queryStream.context();
+
+                switch (queryStream.context()) {
+                    case VARIABLE:
+                        if (queryStream instanceof Intersection) {
+                            this.context = StreamContext.PRESENT;
+                            break;
+                        }
+                    case DELAYED:
+                        if (queryStream instanceof Union) {
+                            // all terms in a top level union must be executable
+                            this.context = StreamContext.ABSENT;
+                            this.itr = Collections.emptyIterator();
+                            return itr;
+                        }
+                    default:
+                        context = queryStream.context();
```

Comment:
Revaluate this switch statement. The VARIABLE case will fall through to the DELAYED case. Not sure if that is the intent or not.